<launch>
    <arg name="MAP_FRAME" default="map" />
    <arg name="CAMERA_FRAME" default="zed2_left_camera_frame" />
    <arg name="BASE_FRAME" default="base_link" />
    <arg name="map_to_odom" default="true"/>
    <arg name="odom_to_base_footprint" default="true"/>
    <arg name="POINT_CLOUD_TOPIC" default="/zed2/zed_node/point_cloud/ds_cloud_registered" />
    <arg name="CLUSTERS_PER_OBJECT" default="3" />
    <arg name="SAVE_IMAGE" default="true" />
    <node 
        pkg="tf" 
        type="static_transform_publisher" 
        name="map_broadcaster" 
        args="0 0 0 0 0 0 1 map odom 20" 
        if="$(arg map_to_odom)"
    />
    <node 
        pkg="tf" 
        type="static_transform_publisher" 
        name="odom_broadcaster" 
        args="0 0 0 0 0 0 1 odom base_footprint 20"
        if="$(arg odom_to_base_footprint)"
    />
    <node name="Detection3DShelves" pkg="object_detector_3d" type="Detection3DShelves" respawn="true" output="screen">
        <param name="POINT_CLOUD_TOPIC" value="$(arg POINT_CLOUD_TOPIC)"/>
        <param name="MAP_FRAME" value="$(arg MAP_FRAME)" />
        <param name="BASE_FRAME" value="$(arg BASE_FRAME)" />
        <param name="CAMERA_FRAME" value="$(arg CAMERA_FRAME)" />
        <param name="USING_CLUSTERING" value="false" />
    </node>
    <node name="Detection3DShelvesClient" pkg="object_detector_3d" type="Detection3DShelvesClient" respawn="true" output="screen" />
</launch>
